src/lib.rs

```diff
@@ -312,19 +334,33 @@ const ERROR_CATEGORY_CANNOT_BE_UNARCHIVED_WHEN_DELETED: &str =
 use system;
 use system::{ensure_root, ensure_signed};
 
-/// Represents a user in this forum.
-#[derive(Debug, Copy, Clone)]
+/// Represents a forum user identifier, it is immutable.
+pub type ForumUserId = u64;
+
+/// Represents a user's information in this forum.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize, Debug))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq)]
 pub struct ForumUser<AccountId> {
     /// Identifier of user
     pub id: AccountId, // In the future one could add things like
-                       // - updating post count of a user
-                       // - updating status (e.g. hero, new, etc.)
-                       //
+    // - updating post count of a user
+    // - updating status (e.g. hero, new, etc.)
+    //
+    pub name: Vec<u8>,
```

Comment:
I will add comments for the variables. My idea is forum has its own account system, use forum user id as the key. Like most of forum, each forum user has his/her name and self-introduction. For membership module, I am not sure about the relationship between forum and  membership since I am not involved in the whole runtime design before. But considering the scenario like a membership could register multiple forum user, one publish thread, others give response. You are right, maybe we need add constraint about name should be unique. Let leave it, I will keep it in my TODO list.